hx711/hx711.go

```diff
@@ -0,0 +1,407 @@
+package hx711
+
+import (
+	"errors"
+	"strconv"
+	"sync"
+	"time"
+
+	"tinygo.org/x/drivers"
+)
+
+type GainAndChannelCfg int
+
+const (
+	None    GainAndChannelCfg = 0 // channel is not used
+	A128    GainAndChannelCfg = 1 // channel A, gain factor 128 - 25 pulses
+	B32     GainAndChannelCfg = 2 // channel B, gain factor 32 - 26 pulses
+	A64     GainAndChannelCfg = 3 // channel A, gain factor 64 - 27 pulses
+	A128B32 GainAndChannelCfg = 4 // channel A@128 and channel B@32 - after first read 26 pulses, after second 25 pulses
+	A64B32  GainAndChannelCfg = 5 // channel A@64 and channel B@32 - after first read 26 pulses, after second 27 pulses
+)
+
+const (
+	defaultResetDuration = 100 * time.Microsecond
+	defaultSettlingTime  = 400 * time.Millisecond // according the data sheet: RATE=0 (10Hz), otherwise use 50 ms
+	// readCycleTime = 100 * time.Millisecond // for RATE=0 (10Hz), otherwise 12.5 ms (80Hz)
```

Comment:
this is not used internally, e.g. for blocking fast updates, but a caller could have a benefit if we add a "MaxCycleTime()" function to the drivers.Sensor interface